ui/src/components/connections/ConnectionDetails.tsx

```diff
@@ -0,0 +1,166 @@
+import { useEnvContext } from "src/contexts/env";
+import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
+import { IDENTIFIER, ISSUE_CREDENTIAL } from "src/utils/constants";
+import { formatDate } from "src/utils/forms";
+import { AsyncTask, isAsyncTaskDataAvailable, isAsyncTaskStarting } from "src/utils/types";
+
+export function ConnectionDetails() {
+  const env = useEnvContext();
+
+  const [connection, setConnection] = useState<AsyncTask<Connection, APIError>>({
+    status: "pending",
+  });
+
+  const { connectionID } = useParams();
+
+  const obfuscateDID = (did: string) => {
+    const didSplit = did.split(":").slice(0, -1).join(":");
+    const address = did.split(":").pop();
+
+    return address
+      ? `${didSplit}${address.substring(0, 5)}...${address.substring(address.length - 5)}`
+      : "-";
+  };
+
+  const fetchConnection = useCallback(
+    async (signal: AbortSignal) => {
+      if (connectionID) {
+        setConnection({ status: "loading" });
+        const response = await getConnection({
+          env,
+          id: connectionID,
+          signal,
+        });
+        if (response.isSuccessful) {
+          setConnection({
+            data: response.data,
+            status: "successful",
+          });
+        } else {
+          if (!isAbortedError(response.error)) {
+            setConnection({ error: response.error, status: "failed" });
+          }
+        }
+      } else {
+        setConnection({ error: { message: "No ConnectionID" }, status: "failed" });
+      }
+    },
+    [connectionID, env]
+  );
+
+  useEffect(() => {
+    const { aborter } = makeRequestAbortable(fetchConnection);
+
+    return aborter;
+  }, [fetchConnection]);
+
+  // TODO PID-481
+  const credentialsList = []; /* isAsyncTaskDataAvailable(connection) && connection.data.credentials
+      ? connection.data.credentials
+      : []; */
+
+  return (
+    <SiderLayoutContent
+      description="View connection information, credential attribute data. Revoke and delete issued credentials."
+      showBackButton
+      showDivider
+      title="Connection details"
+    >
+      <Space direction="vertical" size="large">
+        <Card>
+          {(() => {
+            switch (connection.status) {
+              case "pending":
+              case "loading": {
+                return <LoadingResult />;
+              }
+              case "failed": {
+                return <ErrorResult error={connection.error.message} />;
+              }
+              case "successful":
+              case "reloading": {
+                return (
+                  <Space direction="vertical" size="middle">
+                    <Row align="middle" justify="space-between">
+                      <Card.Meta title="Connection" />
+                      <Button danger icon={<IconTrash />} type="text">
+                        Delete connection
+                      </Button>
```

Comment:
Shouldn't these buttons be consistently styled?

![delete-connection](https://user-images.githubusercontent.com/15896177/229451843-9c04b736-8eb2-4ad1-97c0-2f8f50849ba1.gif)

Reply:
@amonsosanz the dropdown is a link as opposed to the button. It's correct within the system, they are different modalities.

Reply:
ok...